cf/data/data.py

```diff
@@ -9376,16 +9378,16 @@ def stats(
             if all or locals()[stat]:
                 func = getattr(self, stat)
                 if stat in no_weights:
-                    value = func(squeeze=True)
+                    value = delayed(func)(squeeze=True)
                 else:
-                    value = func(squeeze=True, weights=weights)
+                    value = delayed(func)(squeeze=True, weights=weights)
 
                 out[stat] = value
 
         if all or sample_size:
-            out["sample_size"] = int(self.sample_size())
+            out["sample_size"] = delayed(lambda: int(self.sample_size()))()
```

Comment:
I know lambdas are frowned upon somewhat, but it seems intuitive to use one here instead of defining a trivial function, I think.